src/formatDate.js

```diff
@@ -9,6 +9,33 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const result = Array(3).fill(0);
+  const fromSeparator = date.match(/[^\w\s]/);
+
+  const toSeparator = toFormat[toFormat.length - 1].match(/[^\w\s]/)
+    ? toFormat[toFormat.length - 1]
+    : fromSeparator;
+  const dateArray = date.split(fromSeparator);
+  const day = dateArray[fromFormat.indexOf('DD')];
+  const month = dateArray[fromFormat.indexOf('MM')];
+  const yearFromFormat = fromFormat.indexOf('YY') === -1 ? 'YYYY' : 'YY';
+  const year = dateArray[fromFormat.indexOf(yearFromFormat)];
+  const yearToFormat = toFormat.indexOf('YY') === -1 ? 'YYYY' : 'YY';
+  let formatedYear = year;
+
+  if (yearToFormat.length < yearFromFormat.length) {
+    formatedYear = year.slice(2);
+  }
+
+  if (yearToFormat.length > yearFromFormat.length) {
+    formatedYear = +year < 30 ? '20' + year : '19' + year;
```

Comment:
The logic for determining the century prefix for the year assumes that any year less than 30 belongs to the 21st century. This might not be a valid assumption for all use cases. Consider making this logic more flexible or configurable.